ReSharper.FSharp/src/FSharp.Psi.Features/src/PostfixTemplates/CopyAndUpdatePostfixTemplate.fs

```diff
@@ -0,0 +1,197 @@
+        (fun (inner: string) -> $"""{{ {pref} with {propName} = {inner} }}""")
+
+    /// createCopyAndUpdate ["Customer"; "Phone" ; "Number"] -> "{ Customer with Phone = { Customer.Phone with Number =  } }"
+    let rec createCopyAndUpdate (identifiers: string list) =
+        let rec loop
+            (innerString: string)
+            (usedIdentifers: string list)
+            (identifiers: string list)
+            =
+            match identifiers with
+            | []
+            | [ _ ] -> innerString
+            | _ ->
+                let reference = usedIdentifers @ [ identifiers[0] ]
+                let property = identifiers[1]
+
+                let recursivepart =
+                    identifiers[1 .. identifiers.Length - 1]
+
+                innerWith reference property (loop innerString reference recursivepart)
+
+        loop "" [] identifiers
+
+
+    type RecordExprContext =
+        {
+            OuterRecord: IReferenceExpr option
+            Fields: string list
+        }
+        static member Default = { OuterRecord = None; Fields = [] }
+
+    let fieldBelongsToRecord (fsf: FSharpField) =
+        fsf.DeclaringEntity
+        |> Option.map (fun f -> f.IsFSharpRecord)
+        |> Option.defaultValue false
+
+    /// Gets all record qualifiers, returns None if not a record field chain
+    let rec tryGetContext (refExpr: IReferenceExpr) =
+        let rec loop (acc: RecordExprContext) (refExpr: IReferenceExpr) =
+            match refExpr.Qualifier with
+            | null -> Some acc
+            | :? IReferenceExpr as prev ->
+                match prev.Reference.GetFcsSymbol() with
+                | :? FSharpField as field ->
+                    match fieldBelongsToRecord field with
+                    | true -> loop { acc with Fields = field.Name :: acc.Fields } prev
+                    | false -> None // found union case instead
+                | :? FSharpMemberOrFunctionOrValue as outer ->
+                    match acc.OuterRecord with
+                    // would have multiple function calls, don't want that
+                    // e.g { r() with r1 = { r().r1 with {caret} }
+                    | _ when outer.IsFunction -> None
+                    // already have the outer qualified name
+                    | Some _ -> Some acc
+                    | _ -> loop { acc with OuterRecord = Some prev } prev
+                | _ -> None
+            | _ -> None
+
+        loop RecordExprContext.Default refExpr
+
+    /// Gets the innermost record expr in the generated recursive record expr
+    let rec getInnermostExpr (recordExpr: IRecordExpr) =
+        match recordExpr.FieldBindingList with
+        | null -> recordExpr
+        | bindingList ->
+            let innerChildren =
+                // there is only one child in bindingList
+                bindingList
+                    .Children()
+                    .First()
+                    .As<IRecordFieldBinding>()
+                    .Children()
```

Comment:
Please use existing accessors instead of walking the tree manually, e.g. `IRecordLikeExpr.FieldBindings` and `IRecordFieldBinding.Expression`.